catalyst-toolbox-cli/src/cli/rewards/voters.rs

```diff
@@ -0,0 +1,123 @@
+use super::Error;
+use jcli_lib::jcli_lib::block::Common;
+
+use structopt::StructOpt;
+
+use chain_addr::{Discrimination, Kind};
+use chain_impl_mockchain::vote::CommitteeId;
+use jormungandr_lib::interfaces::{Address, Block0Configuration, Initial};
+use std::collections::{HashMap, HashSet};
+use std::ops::{Div, Mul};
+
+const ADA_TO_LOVELACE_FACTOR: u64 = 1_000_000;
+
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub struct VotersRewards {
+    #[structopt(flatten)]
+    common: Common,
+    /// Reward (in LOVELACE) to be distributed
+    #[structopt(long = "total-rewards")]
+    total_rewards: u64,
+}
+
+fn calculate_stake<'address>(
+    committee_keys: &HashSet<Address>,
+    block0: &'address Block0Configuration,
+) -> (u64, HashMap<&'address Address, u64>) {
+    let mut total_stake: u64 = 0;
+    let mut stake_per_voter: HashMap<&'address Address, u64> = HashMap::new();
+
+    for fund in &block0.initial {
+        match fund {
+            Initial::Fund(fund) => {
+                for utxo in fund {
+                    if !committee_keys.contains(&utxo.address) {
+                        let value: u64 = utxo.value.into();
+                        total_stake += value;
+                        let entry = stake_per_voter.entry(&utxo.address).or_default();
+                        *entry += value;
+                    }
+                }
+            }
+            Initial::Cert(_) => {}
+            Initial::LegacyFund(_) => {}
+        }
+    }
+    (total_stake, stake_per_voter)
+}
+
+/// Rewards are u64 for keeping the it we would calculate the inverse total_stake/voter_stake
+fn calculate_inverse_reward_share<'address>(
+    total_stake: u64,
```

Comment:
maybe use `Value` instead of `u64`? Or maybe some different newtype (`Lovelaces`?), although this is nitpicking too.

Reply:
Yeah, I could use the value wrapper for this. Would make sense.